### PR TITLE
Pretty Print Generic EntityType Names

### LINKED ISSUE #1986: Generic EntityTypes have different names across runtimes
Given the following model...
```C#
class Entity<TKey>
{
    public TKey Id { get; set; }
}

class Context : DbContext
{
    public DbSet<Entity<int>> Entities { get; set; }
}
```
...the name of the entity type will be different depending on the platform.

Runtime | Entity Type Name
------- | ----------------
dnx451 | Entity`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=**b77a5c561934e089**]]
dnxcore50 | Entity`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=**7cec85d7bea7798e**]]

Since these are *our* metadata names, we should prettify them to something like `Entity<string>`.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #1986

### CODE REVIEW

**Comment by lajones** [comment]
Looks good. :shipit:

File: test/EntityFramework.Core.Tests/Utilities/TypeNameHelperTest.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+namespace Microsoft.Data.Entity.Tests.Utilities
+{
+    public class TypeNameHelperTest
+    {
+        [Fact]
+        public void Can_pretty_print_CLR_full_name()
+        {
+            // Predefined Types
+            Assert.Equal("System.Int32",
```
Comment (bricelam) [inline test/EntityFramework.Core.Tests/Utilities/TypeNameHelperTest.cs]: Can we replace built-in types? They're:
- bool
- byte
- char
- decimal
- double
- float
- int
- long
- object
- sbyte
- short
- string
- uint
- ulong
- ushort

Comment (smitpatel) [inline test/EntityFramework.Core.Tests/Utilities/TypeNameHelperTest.cs]: Updated.

**Comment by Eilon** [comment]
Looks like we've now significantly diverged the two "pretty print type" implementations. Can we share the code between the two projects by using the Common repo and shared source?

File: src/EntityFramework.Core/Utilities/TypeNameHelper.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using JetBrains.Annotations;
+
+namespace Microsoft.Data.Entity.Utilities
+{
+    public static class TypeNameHelper
+    {
+        private static readonly Dictionary<Type, string> builtInTypeNames = new Dictionary<Type, string>
```
Comment (bricelam) [inline src/EntityFramework.Core/Utilities/TypeNameHelper.cs]: Section 3.4, paragraph 2 of the official [ASP.NET 5 Project Coding Guidelines](https://github.com/aspnet/Home/wiki/Engineering-guidelines#coding-guidelines) states that, "All fields MUST begin with an underscore."

Ok, so it actually doesn't... :grinning: but could you make this field start with an underscore to be consistent with the rest of our code base?

**Comment by bricelam** [comment]
:shipit: Those are some mighty pretty type names.

File: test/EntityFramework.Core.Tests/Utilities/TypeNameHelperTest.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            Assert.Equal("long", TypeNameHelper.GetTypeDisplayName(typeof(long)));
+            Assert.Equal("object", TypeNameHelper.GetTypeDisplayName(typeof(object)));
+            Assert.Equal("sbyte", TypeNameHelper.GetTypeDisplayName(typeof(sbyte)));
+            Assert.Equal("short", TypeNameHelper.GetTypeDisplayName(typeof(short)));
+            Assert.Equal("string", TypeNameHelper.GetTypeDisplayName(typeof(string)));
+            Assert.Equal("uint", TypeNameHelper.GetTypeDisplayName(typeof(uint)));
+            Assert.Equal("ulong", TypeNameHelper.GetTypeDisplayName(typeof(ulong)));
+            Assert.Equal("ushort", TypeNameHelper.GetTypeDisplayName(typeof(ushort)));
+        }
+        class A { }
```
Comment (bricelam) [inline test/EntityFramework.Core.Tests/Utilities/TypeNameHelperTest.cs]: Oh, can you specify a visibility modifier on these nested types too? (`private` is fine)

**Comment by smitpatel** [comment]
created aspnet/common#15

**Comment by smitpatel** [comment]
Files #2054 & aspnet/Logging#175

**Pull request merged**